From 23fd5e8d0ca23d96c77b78b5d7f47c706aac4668 Mon Sep 17 00:00:00 2001
From: IAP Desktop Build <iap-desktop+build@google.com>
Date: Thu, 13 May 2021 12:12:51 +0200
Subject: [PATCH 3/3] Translate Enter to CR instead of LF

Enter keys were previously translated to LF, which
caused some programs (such as nano) to fail.
---
 VtNetCore/VirtualTerminal/KeyboardTranslations.cs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/VtNetCore/VirtualTerminal/KeyboardTranslations.cs b/VtNetCore/VirtualTerminal/KeyboardTranslations.cs
index 25131c6..7310782 100644
--- a/VtNetCore/VirtualTerminal/KeyboardTranslations.cs
+++ b/VtNetCore/VirtualTerminal/KeyboardTranslations.cs
@@ -151,8 +151,8 @@
             // | Z      | z      | Z      | \x1a     |
             { "Back",    new KeyboardTranslation { Normal = "\u007F", Shift = "\b", Control = "\u007F" } },
             { "Tab",     new KeyboardTranslation { Normal = "\t", Shift = "\u001b[Z" } },
-            { "Enter",   new KeyboardTranslation { Normal = "\n", Shift = "\n", Control = "\n" } },
-            { "Return",  new KeyboardTranslation { Normal = "\n", Shift = "\n", Control = "\n" } },
+            { "Enter",   new KeyboardTranslation { Normal = "\r", Shift = "\r", Control = "\r" } },
+            { "Return",  new KeyboardTranslation { Normal = "\r", Shift = "\r", Control = "\r" } },
             { "Escape",  new KeyboardTranslation { Normal = "\u001b\u001b", Shift = "\u001b\u001b", Control = "\u001b\u001b" } },
             { "A",       new KeyboardTranslation { Control = "\u0001" } },
             { "B",       new KeyboardTranslation { Control = "\u0002" } },
-- 
2.17.1.windows.2

